siphon/http_util.py

```diff
@@ -0,0 +1,476 @@
+import posixpath
+from collections import OrderedDict
+from datetime import datetime, timedelta, tzinfo
+from itertools import chain
+try:
+    from urllib.parse import urlencode, urljoin  # noqa
+except ImportError:
+    from urllib import urlencode
+    from urlparse import urljoin  # noqa
+
+import requests
+
+from io import BytesIO
+from . import __version__
+
+user_agent = 'Siphon (%s)' % __version__
+
+HTTPError = requests.HTTPError
+
+
+class BadQueryError(Exception):
+    r'Exception raised when a query fails.'
+    pass
+
+
+# A UTC class. Taken from standard library docs.
+class UTC(tzinfo):
+    r'''UTC'''
+    ZERO = timedelta(0)
+
+    def utcoffset(self, dt):  # pylint:disable=unused-argument
+        return self.ZERO
+
+    def tzname(self, dt):  # pylint:disable=unused-argument
+        return r'UTC'
+
+    def dst(self, dt):  # pylint:disable=unused-argument
+        return self.ZERO
+
+utc = UTC()
+
+
+def create_http_session():
+    r'''Create a new HTTP session with our user-agent set.
+
+    Returns
+    -------
+    session : ``requests.Session``
+        The created session
+
+    See Also
+    --------
+    urlopen
+    '''
+
+    ret = requests.Session()
+    ret.headers['User-Agent'] = user_agent
+    return ret
+
+
+def urlopen(url, **kwargs):
+    r'''GET a file-like object for a URL using HTTP.
+
+    This is a thin wrapper around ``requests.get`` that returns a file-like object
+    wrapped around the resulting content.
+
+    Parameters
+    ----------
+    url : string
+        The URL to request
+
+    kwargs : arbitrary keyword arguments
+        Additional keyword arguments to pass to ``requests.get``.
+
+    Returns
+    -------
+    fobj : file-like object
+        A file-like interface to the content in the response
+
+    See Also
+    --------
+    http_get
+    '''
+
+    return BytesIO(create_http_session().get(url, **kwargs).content)
+
+
+def parse_iso_date(s):
+    r'''Parse a string containing an ISO-8601 formatted date
```

Comment:
I say do that once we have something else to put in `util` at some point; it no longer exists right now, as all of the old functions in `util` were http-related.